content/docs/guides/authentication/jwt.md

```diff
@@ -276,183 +244,208 @@ Copy the returned JWT string and paste it into the following curl request:
 
 ```swift
-let authHeader = request.headers["Authorization"]
-guard let authComponents = authHeader?.components(separatedBy: " "),
-    authComponents.count == 2,
-    authComponents[0] == "Bearer",
-    let jwt = try? JWT<C>(jwtString: authComponents[1], verifier: App.jwtVerifier)
-else {
-    return completion(nil, .unauthorized)
-}
-completion(TypeSafeJWT(jwt: jwt), nil)
+let authenticationMiddleware = Credentials()
+authenticationMiddleware.register(plugin: jwtCredentials)
+app.router.get("/jwtprotected", middleware: authenticationMiddleware)
+app.router.get("/jwtprotected") { request, response, next in
+    guard let userProfile = request.userProfile else {
+            Log.verbose("Failed raw token authentication")
+            response.status(.unauthorized)
+            try response.end()
+            return
+        }
+    response.send("\(userProfile.id)\n")
+        next()
+    }
 ```
 
-Your completed TypeSafeJWT.swift file should look as follows:
+Using the same terminal commands as earlier, you can test your JWT generation and authentication:
 
-```swift
-import SwiftJWT
-import Kitura
-
-struct TypeSafeJWT<C: Claims>: TypeSafeMiddleware {
-    let jwt: JWT<C>
-    static func handle(request: RouterRequest, response: RouterResponse, completion: @escaping (TypeSafeJWT?, RequestError?) -> Void) {
-        let authHeader = request.headers["Authorization"]
-        guard let authComponents = authHeader?.components(separatedBy: " "),
-            authComponents.count == 2,
-            authComponents[0] == "Bearer",
-            let jwt = try? JWT<C>(jwtString: authComponents[1], verifier: App.jwtVerifier)
-        else {
-            return completion(nil, .unauthorized)
-        }
-        completion(TypeSafeJWT(jwt: jwt), nil)
-    }
-}
+```
+curl -X GET \
+http://localhost:8080/jwtprotected \
+-H 'X-Token-Type: JWT' \
+-H 'Authorization: Bearer <Your JWT string here>'
 ```
 
----
+Congratulations! You are now using your own set of custom claims to generate a JWT and authenticate a user!
+
+##Step 10: JWTs on Codable Routes (Optional)
+
+In our example we used raw routing since we chose to pass the user credentials via the request headers. If we want to use JWTs on our codable routes, we need to encapsulate the verification and creation of the users JWT in a TypeSafeMiddleware. We can then register our TypeSafeMiddleware on a Codable route to authenticate the user and access their claims.
 
-###Step 9b: Register TypeSafeJWT on a route.
+###Step 10a: Register TypeSafeJWT on a route.
 
-Back in our routes file, JWTRoutes.swift, we are going to register a new route using our TypeSafeMiddleware.
+In the JWTRoutesFile.swift we are going to create a new route for a TypeSafeJWT.
 
-In the function initializeJWTRoutes, add a new route called "/jwtCodable"
+In the function initializeJWTRoutes, declare the verifier for the TypeSafeJWT (this example is using the hs256 algorithm):
```

Comment:
we should refer to this as the HS256 algorithm can you check it's all uppercase?